### PR TITLE
autogen: fix patchelf invocation to remove generic build path from rpath

### PR DESCRIPTION
**Pull request by siscia**
Refers to #106165 

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by ztzg — APPROVED** [review]
LGTM.

(Hit this when rebuilding a number of packages on RHEL with sandboxing disabled.)

**Comment by Dridus** [comment]
LGTM. this resolved an issue I had with cross compiling to armv7 on a WSL+Debian+Nix build platform

**Comment by cpakkala** [comment]
Had the same problem and this fixed it.  I think we, who have to compile our systems (non-standard nix dir location) without sandboxing (work permission restrictions; no I can't use proot or the like; big brother doesn't allow user namespace in kernel) are, and forever will be, the red-headed step-children.

File: pkgs/development/tools/misc/autogen/default.nix [hunk 1]
```diff
@@ -79,7 +79,7 @@ stdenv.mkDerivation rec {
   '' + stdenv.lib.optionalString (!stdenv.hostPlatform.isDarwin) ''
     # remove /build/** from RPATHs
```
Comment (r-burns) [inline pkgs/development/tools/misc/autogen/default.nix:80]:
```suggestion
    # remove build directory (/build/**, or /tmp/nix-build-**) from RPATHs
```
might make it a little more clear why this change was needed

**Review by r-burns — COMMENTED** [review]
Thanks! This fixes the linked issue for me. Agree with Sandro that this should be rebased onto staging due to the quantity of linux rebuilds. Otherwise LGTM.

> I think we, who have to compile our systems [...] are, and forever will be, the red-headed step-children.

While I certainly wouldn't expect first-class support, I think there are enough of us out there that if we stick up for each other and nip these issues in the bud, we'll have a pretty good user experience. From what I've seen, issues like this are rare and fairly easy to patch around.

**Comment by siscia** [comment]
Guys please don't expect me to work on this.

If somebody want to take charge of this PR, please go ahead with any means, it just won't be me.

**Comment by r-burns** [comment]
Yeah no worries, thanks for sharing this patch. Continued in https://github.com/NixOS/nixpkgs/pull/126557
